### ISSUE TITLE
In WPF at mousewheel zoom-out the Map sticks to the mousepointer

### ISSUE THREAD

**Issue by pauldendulk** [Mapsui/Mapsui#908]
What happens is that if you zoom out with the mouse wheel and at the same time mouse the mouse outside of the edge of the visible map the map will move along with the mouse as if it is glued to it.

**Comment by pauldendulk** [comment]
### The problem

This was hard to debug. See the code below. There are several steps to it. What this code did not anticipate is that the _viewport here is the LimitedViewport and it will limit the already at the first SetCenter which messes up the logic. 

```csharp
        public void ZoomTo(double resolution, Point centerOfZoom)
        {
            // 1) Temporarily center on the center of zoom
            _viewport.SetCenter(_viewport.ScreenToWorld(centerOfZoom));

            // 2) Then zoom 
            _viewport.SetResolution(resolution);

            // 3) Then move the temporary center of the map back to the mouse position
            _viewport.SetCenter(_viewport.ScreenToWorld(
                _viewport.Width - centerOfZoom.X,
                _viewport.Height - centerOfZoom.Y));

            Navigated?.Invoke(this, EventArgs.Empty);
        }
```

### The Fix
To solve this I added this logic to the Viewport itself so that from here only _viewport.SetResolution(resolution, centerOfZoom) needs to be called.

### What could be improved
To solve this some more logic needed to be added to the Viewport which I would rather keep simple. Perhaps the layering of Viewport and LimitedViewport is not necessary anymore now that the Navigator is added. Perhaps the LimitingViewport should be deleted and the ViewportLimiter should be called from the Navigator.

**Comment by pauldendulk** [comment]
Fixed with #909
